### PR TITLE
refactor: separate v1 and v2 (VE)

### PR DESCRIPTION
**Pull request by alexanderbez**
As we introduce further improvements and changes to v1 of POB, while simultaneously needing to maintain the prepped v2 (VE) work while also gearing up to work on blockbuster, it's become a big pain point maintaining and testing all of these domains.

What this PR does is the following:

- Move all VE implementation into `abci/v2`. We will continue to maintain this while also keeping existing tests for it. Once the v0.48 SDK is released, we can revert this.
- Move existing v1 to `abci/` and use this for E2E.

TL;DR: This is to avoid the backwards PR against `release/v1.x.x` -> backport to main -> fix conflicts

### CODE REVIEW

**Comment by davidterpay** [comment]
This makes sense to me, would it make sense to just move the V1 into its own folder? so the file structure is `abci/v1` and `abci/v2`?

**Review by davidterpay — APPROVED** [review]

**Comment by alexanderbez** [comment]
@Mergifyio backport release/v1.x.x-polaris
